src/Confluent.Kafka/Timestamp.cs

```diff
@@ -36,16 +36,38 @@ public struct Timestamp
         /// <summary>
         ///     Initializes a new instance of the Timestamp structure.
         /// </summary>
+        /// <param name="unixTimestampMs">
+        ///     The unix millisecond timestamp.
+        /// </param>
+        /// <param name="type">
+        ///     The type of the timestamp.
+        /// </param>
+        public Timestamp(long unixTimestampMs, TimestampType type)
+        {
+            Type = type;
+            UnixTimestampMs = unixTimestampMs;
+        }
+
+        /// <summary>
+        ///     Initializes a new instance of the Timestamp structure.
+        ///     Prefer the (long, TimestampType) overload if possible.
```

Comment:
To be more clear : 
- Timestamp struct are created only from librdkafka return, so remove the constructor with datetime is ok.
- Message (returned by consumer and delivery handler) contains TimeStamp, so I would keep the datetime property for easy usage
- ProduceRecord contains (will contain, I couldn't do both in atomic PR) a DateTime and UnixTimestampMs set property

What I don't like about dateTime in .NET is this:
if(DateTime.Now.AddMinutes(1) > timestamp.DateTime)
return unexpected value when kind is not the same (depend on timezone) I know it's user side to be sure he uses correct kind on both side, but DateTimeOffset resolve this (and it has DateTime / DateTimeUtc property). We  could do the same with a LocalDateTime, UtcDateTimeUtc property here. Microsoft actually advise using DateTimeOffset for utc datetime


https://msdn.microsoft.com/library/bb384267(v=vs.110).aspx
> These uses for DateTimeOffset values are much more common than those for DateTime values. As a result, DateTimeOffset should be considered the default date and time type for application development.